assets/js/base/components/quantity-selector/stories/index.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { Story, Meta } from '@storybook/react';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import QuantitySelector, { QuantitySelectorProps } from '..';
+
+export default {
+	title: 'WooCommerce Blocks/@base-components/QuantitySelector',
+	component: QuantitySelector,
+	args: {
+		itemName: 'widgets',
+	},
+	argTypes: {
+		quantity: { control: { disable: true } },
+	},
+} as Meta< QuantitySelectorProps >;
+
+const Template: Story< QuantitySelectorProps > = ( args ) => {
+	const [ quantity, setQuantity ] = useState( 1 );
+
+	const onChange = ( newVal: number ) => {
+		args.onChange?.( newVal );
```

Comment:
I didn't know the optional chaining can be used as a check for a function here, that's cool!